src/NHSD.BuyingCatalogue.SolutionLists.Application/Domain/SolutionList.cs

```diff
@@ -1,32 +1,30 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using NHSD.BuyingCatalogue.SolutionLists.Application.Persistence;
 using NHSD.BuyingCatalogue.SolutionLists.Contracts.Persistence;
 
 namespace NHSD.BuyingCatalogue.SolutionLists.Application.Domain
 {
     internal sealed class SolutionList
     {
-        public List<SolutionListItem> Solutions { get; }
-
         internal SolutionList(ISet<Guid> capabilityIdList, IEnumerable<ISolutionListResult> solutionListResults)
         {
-            var solutions = solutionListResults.Select(s => new SolutionListItem(s)).ToArray().Distinct(new SolutionListItemComparer());// force eager execution
-
-            foreach (var item in solutionListResults)
+            var solutions = new Dictionary<string, SolutionListItem>();
+            foreach (var result in solutionListResults)
             {
-                solutions.Single(s => s.Id == item.SolutionId).Capabilities.Add(new SolutionListItemCapability(item));
+                if (!solutions.ContainsKey(result.SolutionId))
+                {
+                    solutions.Add(result.SolutionId, new SolutionListItem(result));
+                }
+
+                solutions[result.SolutionId].Capabilities.Add(new SolutionListItemCapability(result));
             }
 
-            Solutions = CapabilityFilter(capabilityIdList, solutions).ToList();
+            Solutions = solutions.Values
+                .Where(s => s.Capabilities.Select(c=>c.Id).Intersect(capabilityIdList).Count() == capabilityIdList.Count )
+                .ToList();
         }
 
-        private IEnumerable<SolutionListItem> CapabilityFilter(ISet<Guid> capabilityIdList, IEnumerable<SolutionListItem> solutions)
-        {
-            return capabilityIdList.Any() ?
-                solutions.Where(solution => capabilityIdList.Intersect(solution.Capabilities.Select(capability => capability.Id)).Count() == capabilityIdList.Count()) :
-                solutions;
-        }
+        public List<SolutionListItem> Solutions { get; }
```

Comment:
Could we move this property back to the top? Initially I tried to be consistent with properties at the top of a class

Reply:
reordered & applied change to dotsettings file.